LayoutTest/TestCase/LYTLayoutFailingTestSnapshotRecorder.h

```diff
@@ -0,0 +1,27 @@
+// © 2016 LinkedIn Corp. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+#import <XCTest/XCTest.h>
+
+@interface LYTLayoutFailingTestSnapshotRecorder : NSObject<XCTestObservation>
+
+@property (nonatomic, strong) UIView *viewUnderTest;
```

Comment:
They need to be set from the LYTLayoutTestCase to keep track of the currently failing view. When 
`- (void)recordFailureWithDescription:(NSString *)description inFile:(NSString *)filePath atLine:(NSUInteger)lineNumber expected:(BOOL)expected {`
is called we don't have a reference to the view and data that failed. Options are to leave as is, change it to a method to set them rather than having them exposed or save the view and data in the LYTLayoutTestCase and pass them into saveImageOfCurrentViewWithInvocation when we get a failure?